### DISCUSSION TITLE
```using``` IDisposable[]

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7772] [Language Ideas]
It would be really nice if ```using``` could target an array/enumerable of IDisposable and dispose everything in it.

**Comment by HaloFour** [comment]
```cs
using var _ = new CompositeDisposable([ disposable1, disposable2, disposable3 ]);
```

**Reply by TonyValenti** [reply]
Ahh!  Good idea.  That class should be moved from RX to the core!

**Reply by KennethHoff** [reply]
In the [`Modernizing Rx.NET | .NET Conf 2023`](https://youtu.be/hLjqf4PmG9E?si=TqDaZvdT3ZnV468d&t=1440) talk (at 24:00) you can see there is some plan about merging some Rx.Net collection-y things into the .Net Runtime libraries, so maybe this might be a part of it?
